File: employee-management/src/main/java/com/dreamteam/employeemanagement/model/UserProject.java
package com.dreamteam.employeemanagement.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.UUID;
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "user_project")
public class UserProject {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    @ManyToOne
    @JoinColumn(name = "accountId")
    private Account user;

    @ManyToOne
    @JoinColumn(name = "projectId")
    private Project project;
    @Column(name = "startDate")
    private LocalDateTime startDate;
    @Column(name = "endDate")
    private LocalDateTime endDate;
    @Column(name = "description")
    private String description;
    }

File: employee-management-app/src/app/pages/profile/models/userProject.ts
import {FormControl, Validators} from "@angular/forms";
import {Designation} from "../../register/models/Designation";
import {UserProfile} from "./userProfile";
import {Project} from "./project";

export class UserProject {
  id: string;
  user: UserProfile | undefined;
  project: Project;
  startDate: Date;
  endDate: Date;
  description: string;

  constructor(id: string,
              user: UserProfile | undefined,
              project: Project,
              startDate: Date,
              endDate: Date,
              description: string
  ) {
    this.id = id;
    this.user = user;
    this.project = project;
    this.startDate = startDate;
    this.endDate = endDate;
    this.description = description;
  }
}
